app/forms/search_form.rb

```diff
@@ -62,4 +74,54 @@ def apply(scope, programme: nil)
 
     scope.order_by_name
   end
+
+  private
+
+  def handle_session_filters
+    if clear_filters
+      clear_from_session
+    elsif has_filters?
+      store_in_session
+    else
+      load_from_session
+    end
+  end
+
+  def store_in_session
+    return if session.nil?
+
+    if has_filters?
+      session[SESSION_KEY] ||= {}
+      session[SESSION_KEY][path_key] = attributes
+    end
+  end
+
+  def load_from_session
+    return if session.nil? || session[SESSION_KEY].blank?
+
+    stored_filters = session[SESSION_KEY][path_key]
+    return if stored_filters.blank?
+
+    assign_attributes(stored_filters)
```

Comment:
Does this mean that stored attributes will take priority over those passed in as query parameters if this happens at the end of `initialize`?

Reply:
No, `handle_session_filters` will check if any of the attributes are present and if so, overrides the stored filters immediately. `load_from_session` only gets called if all the attributes are blank which would only happen if the user visited the page without any filters.